x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connectors/create_connector/components/manual_configuration.tsx

```diff
@@ -0,0 +1,415 @@
+            </EuiTitle>
+            <EuiSpacer size="m" />
+            <EuiFormRow
+              fullWidth
+              helpText={i18n.translate(
+                'xpack.enterpriseSearch.startStep.euiFormRow.connectorNameLabel',
+                { defaultMessage: 'If no API key is provided, one will be generated for you.' }
+              )}
+              label={i18n.translate(
+                'xpack.enterpriseSearch.startStep.euiFormRow.connectorNameLabel',
+                { defaultMessage: 'API key name' }
+              )}
+            >
+              <EuiFieldText
+                data-test-subj="enterpriseSearchStartStepFieldText"
+                fullWidth
+                name="first"
+                placeholder="Your encoded API key"
+                value={''}
+                onChange={() => {}}
+              />
+            </EuiFormRow>
+            <EuiSpacer size="s" />
+            <EuiLink
+              data-test-subj="enterpriseSearchFlyoutManualConfigContentCreateAnApiKeyLink"
+              href="https://www.elastic.co/guide/en/cloud/current/ec-api-keys.html"
+              target="_blank"
+              external
+            >
+              {i18n.translate('xpack.enterpriseSearch.manualConfiguration.createAPIKey', {
+                defaultMessage: 'Manage API key',
+              })}
+            </EuiLink>
+          </EuiPanel>
+        </EuiFlexItem>
+      </EuiFlyoutBody>
+      <EuiFlyoutFooter>
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              data-test-subj="enterpriseSearchFlyoutManualConfigContentCloseButton"
+              iconType="cross"
+              onClick={() => setIsFlyoutVisible(false)}
+              flush="left"
+            >
+              {i18n.translate(
+                'xpack.enterpriseSearch.flyoutManualConfigContent.closeButtonEmptyLabel',
+                { defaultMessage: 'Close' }
+              )}
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              data-test-subj="enterpriseSearchFlyoutManualConfigContentSaveButton"
+              onClick={() => {
+                setIsFlyoutVisible(false);
+                setIsNextStepEnabled(true);
+                setTimeout(() => {
+                  window.scrollTo({
+                    behavior: 'smooth',
+                    top: window.innerHeight,
+                  });
+                }, 100);
+              }}
+              fill
+            >
+              {i18n.translate(
+                'xpack.enterpriseSearch.flyoutManualConfigContent.saveConfigurationButtonLabel',
+                { defaultMessage: 'Save configuration' }
+              )}
+            </EuiButton>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlyoutFooter>
+    </>
+  );
+  const [flyoutConten, setFlyoutConten] = useState<React.FC>(flyoutManualConfigContent);
```

Comment:
It is a good idea to keep all the hooks `useState`, `useEffect` etc. at the top of the component.

Reply:
Moving this on top I recall why it's there. Passing as a default value `flyoutManualConfigContent` needs to be below it. Perhaps we could just not pass a default value to avoid this issue.

Reply:
Done not setting a default value and moved on top